src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5339,41 +5370,92 @@ void wm_vuldet_feed_update_by_interval(vu_task_ctx_t* current_task, wm_vuldet_t*
 }
 
 void wm_vuldet_scan_by_interval(vu_task_ctx_t* current_task, wm_vuldet_t* vuldet) {
-    int* agents_array = NULL;
-    int i = 0;
     int sock = wm_vuldet_get_wdb_socket();
-    char status_message[OS_SIZE_64] = {0};
+    scan_agent *agents = NULL;
+    scan_agent *f_agent = NULL;
+    scan_agent *agent_aux = NULL;
 
-    // Scaning the manager
-    if (wm_vuldet_collect_agent_info(000, vuldet)) {
-        mterror(WM_VULNDETECTOR_LOGTAG, VU_NO_AGENT_ERROR);
-    } else {
-        snprintf(status_message, OS_SIZE_64, " Agent: manager.");
-        wm_vuldet_create_and_send_message(current_task, WM_TASK_IN_PROGRESS, status_message);
-        if (wm_vuldet_run_scan(vuldet)) {
-            mterror(WM_VULNDETECTOR_LOGTAG, VU_AG_CHECK_ERR);
+    // Collecting agents data
+    int i = -1;
+    int* agents_array = wdb_get_agents_by_connection_status(AGENT_CS_ACTIVE, &sock);
+
+    do {
+        os_calloc(1, sizeof(scan_agent), agent_aux);
+        if (wm_vuldet_collect_agent_info(i == -1 ? 000 : agents_array[i], vuldet, agent_aux)) {
```

Comment:
Since VU_NO_AGENT_ERROR has the format we can just use (0)

Reply:
A `define` was created to indicate that we are talking about the manager's ID.